frontend/src/components/pages/your_clubs/YourClubs.tsx

```diff
@@ -1,10 +1,113 @@
-import * as React from 'react';
+import * as React from "react";
+import AddIcon from "@material-ui/icons/Add";
+import Button from "@material-ui/core/Button";
+import { ClubList } from "./ClubList";
+import { ClubProps } from
+  "../../../services/mock_backend/mock_your_clubs_backend";
+import { createStyles } from "@material-ui/core/styles";
+import { defaultServices } from "../../contexts/contexts";
+import FormControl from "@material-ui/core/FormControl";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import InputLabel from "@material-ui/core/InputLabel";
+import { makeStyles } from "@material-ui/core/styles";
+import MenuItem from "@material-ui/core/MenuItem";
+import Select from "@material-ui/core/Select";
+import { ServiceContext } from "../../contexts/contexts";
+import { Theme } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      display: 'flex',
+      flexDirection: 'column',
+      flexWrap: 'wrap',
+      marginTop: '100px',
+      marginLeft: theme.spacing(4),
+    },
+    button : {
+      margin: theme.spacing(1),
+      marginLeft: theme.spacing(0),
+      marginTop: '20px',
+      maxHeight: '50px',
+    },
+    formControl: {
+      margin: theme.spacing(1),
+      marginBottom: '5px',
+      marginRight: theme.spacing(8),
+      minWidth: 120,
+    },
+    topUtilitiesContainer: {
+      display: 'flex',
+      justifyContent: 'center',
+    },
+  }),
+);
 
 export const YourClubs = () => {
+  const classes = useStyles();
+  const DEFAULT_NUM_DISPLAYED = 10;
+
+  /**
+   * ServiceHandlers is an object containing various TS Handlers and provides
+   * functionality to communicate data from the frontend to the backend.
+   */
+  const contextServices = React.useContext(ServiceContext);
+  const yourClubsHandlerService = contextServices.yourClubsHandlerService;
+
+  const [listedClubs, setListedClubs] =
+    React.useState<ClubProps[]|undefined>(undefined);
+  const [numClubsDisplayed, setNumClubsDisplayed] =
+    React.useState<number|undefined>(undefined);
```

Comment:
This should be set to DEFAULT_NUM_DISPLAYED, not undefined. That also means you might need to move the React.useEffect part into a function so that you can call it immediately, as well as re-calling it whenever numClubsDisplayed changes

Reply:
I didn't notice any loss in functionality when changing it to DEFAULT_NUM_DISPLAYED; handleNumClubsChange handles it upon changing, which then calls the async function in the React.useEffect. I believe this means I'm clear, but I could be wrong.

Reply:
Ok, guess React runs the effect immediately then. Good to know!